bin/dfx-sns-wait-for-lifecycle

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+set -euo pipefail
+SOURCE_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+PATH="$SOURCE_DIR:$PATH"
+
+print_help() {
+  # ShellCheck somehow thinks this code is unreachable.
+  # shellcheck disable=SC2317
+  cat <<-EOF
+
+	Wait up to 2 minutes for the current SNS to reach the given lifecycle.
+	EOF
+}
+
+LIFECYCLES=(
+  "pending"
+  "open"
+  "committed"
+  "aborted"
+  "adopted"
+)
+
+LIFECYCLE_LIST=$(printf " %s," "${LIFECYCLES[@]}")
+LIFECYCLE_LIST=${LIFECYCLE_LIST%,*}
+
+# Source the clap.bash file ---------------------------------------------------
+source "$SOURCE_DIR/clap.bash"
+# Define options
+clap.define short=l long=lifecycle desc="One of $LIFECYCLE_LIST" variable=LIFECYCLE default="open"
```

Comment:
Not sure this script needs a default lifecycle value, but up to you.

Reply:
Otherwise it needs extra logic to handle an empty lifecycle.
Seems harmless like this.